packages/web-components/fast-foundation/src/dialog/dialog.ts

```diff
@@ -196,28 +196,32 @@ export class Dialog extends FoundationElement {
             return;
         }
 
-        if (e.shiftKey && e.target === bounds[0]) {
+        const composed: EventTarget[] = e.composedPath();
+
+        if (e.shiftKey && composed.includes(bounds[0])) {
             bounds[bounds.length - 1].focus();
             e.preventDefault();
-        } else if (!e.shiftKey && e.target === bounds[bounds.length - 1]) {
+        } else if (!e.shiftKey && composed.includes(bounds[bounds.length - 1])) {
             bounds[0].focus();
             e.preventDefault();
         }
 
         return;
     };
 
-    private getTabQueueBounds = (): (HTMLElement | SVGElement)[] => {
-        const bounds: HTMLElement[] = [];
-
-        return Dialog.reduceTabbableItems(bounds, this);
+    private getTabQueueBounds = (): FocusableElement[] => {
+        return tabbable(this, {
+            getShadowRoot: () => {
+                return undefined;
+            },
+        });
     };
 
     /**
      * focus on first element of tab queue
      */
     private focusFirstElement = (): void => {
-        const bounds: (HTMLElement | SVGElement)[] = this.getTabQueueBounds();
+        const bounds: FocusableElement[] = this.getTabQueueBounds();
```

Comment:
bounds in handleTabKeyDown() should be FocusableElement[] to follow this pattern